Mathlib/RepresentationTheory/Homological/GroupCohomology/FiniteCyclic.lean

```diff
@@ -0,0 +1,141 @@
+public import Mathlib.RepresentationTheory.Homological.GroupCohomology.LowDegree
+
+/-!
+# Group cohomology of a finite cyclic group
+
+Let `k` be a commutative ring, `G` a group and `A` a `k`-linear `G`-representation. Given
+endomorphisms `φ, ψ : A ⟶ A` such that `φ ∘ ψ = ψ ∘ φ = 0`, denote by `Chains(A, φ, ψ)` the
+periodic chain complex `... ⟶ A --φ--> A --ψ--> A --φ--> A --ψ--> A ⟶ 0` and by
+`Cochains(A, φ, ψ)` the periodic cochain complex
+`0 ⟶ A --ψ--> A --φ--> A --ψ--> A --φ--> A ⟶ ...`.
+
+When `G` is finite and generated by `g : G`, then `P := Chains(k[G], N, ρ(g) - Id)` (with `ρ` the
+left regular representation) is a projective resolution of `k` as a trivial representation.
+In this file we show that for `A : Rep k G`, `Hom(P, A)` is isomorphic to
+`Cochains(A, N, ρ_A(g) - Id)` as a complex of `k`-modules, and hence the cohomology of this complex
+computes group cohomology.
+
+## Main definitions
+
+* `Rep.FiniteCyclicGroup.groupCohomologyIso₀ A g hg`: given a finite cyclic group `G` generated by
+  `g` and a representation `A : Rep k G`, this is an isomorphism `H⁰(G, A) ≅ Ker(ρ_A(g) - Id)`.
+* `Rep.FiniteCyclicGroup.groupCohomologyIsoOdd A g hg i hi`: given a finite cyclic group `G`
+  generated by `g` and a representation `A : Rep k G`, this is an isomorphism between `Hⁱ(G, A)`
+  and the homology of `A --(ρ(g) - Id)--> A --N--> A` for all odd `i`.
+* `Rep.FiniteCyclicGroup.groupCohomologyIsoEven A g hg i hi`: given a finite cyclic group `G`
+  generated by `g`, and a representation `A : Rep k G`, this is an isomorphism between `Hⁱ(G, A)`
+  and the homology of `A --N--> A --(ρ(g) - Id)--> A` for all positive even `i`.
+
+-/
+
+@[expose] public section
+
+universe v u
+
+open CategoryTheory
+
+namespace Rep.FiniteCyclicGroup
+
+variable {k G : Type u} [CommRing k] [CommGroup G] [Fintype G] (A : Rep k G) (g : G)
+  (hg : ∀ x, x ∈ Subgroup.zpowers g)
+
+/-- Given a finite cyclic group `G` generated by `g : G` and a `k`-linear `G`-representation `A`,
+the periodic cochain complex
+`0 ⟶ Hom(k[G], A) --(- ∘ (ρ(g) - 𝟙))--> Hom(k[G], A) --(- ∘ N)--> Hom(k[G], A) ⟶ ...`
+is isomorphic as a complex in `ModuleCat k` to
+`0 ⟶ A --(ρ(g) - 𝟙)--> A --N--> A --(ρ(g) - 𝟙)--> A --N--> A ⟶ ...`. -/
+@[simps!]
+noncomputable def homResolutionIso :
+    (resolution k g hg).complex.linearYonedaObj k A ≅ moduleCatCochainComplex A g :=
+  HomologicalComplex.Hom.isoOfComponents (fun _ => (leftRegularHomEquiv A).toModuleIso) <| by
+    rintro i j ⟨rfl⟩
+    ext (x : leftRegular _ _ ⟶ _)
+    by_cases hi : Even i
+    · have : ¬(Even (i + 1)) := (not_iff_comm.1 Nat.even_add_one.symm).2 hi
+      simp [hi, this, ← hom_comm_apply x]
+    · simp [hi, Nat.even_add_one.2 hi, Representation.norm, ← hom_comm_apply x]
+
+open ShortComplex Limits
+
+/-- Given a finite cyclic group `G` generated by `g` and `A : Rep k G`, `H⁰(G, A)` is isomorphic
+to the kernel of `ρ(g) - Id(A)`. -/
+noncomputable abbrev groupCohomologyIso₀ (hg : ∀ x, x ∈ Subgroup.zpowers g) :
```

Comment:
```suggestion
noncomputable abbrev groupCohomologyIso₀ :
```
and same for all definition below, where `hg` can be picked up from the `variable` at the top (the only thing it changes is the order wrt the `[DecidableEq G]`, but since that one is an instance it’s probably fine?